components/suggestion/generic_channel_provider.tsx

```diff
@@ -47,28 +62,29 @@ class ChannelSuggestion extends Suggestion {
 }
 
 export default class ChannelProvider extends Provider {
-    constructor(channelSearchFunc) {
+    autocompleteChannels: any;
```

Comment:
as you already defined the type above, you can reuse it here
```suggestion
    autocompleteChannels: ChannelSearchFunc;
```